src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private int threshold;
+
+    public static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.table = (Node<K, V>[]) new Node[capacity];
+        this.size = 0;
+        this.threshold = (int) (capacity * LOAD_FACTOR);
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : key.hashCode() & Integer.MAX_VALUE; // Безопасный хэш
```

Comment:
The hash function uses `key.hashCode() & Integer.MAX_VALUE`, which is a common way to ensure a non-negative hash code. However, consider using `Math.abs(key.hashCode())` to avoid potential issues with `Integer.MIN_VALUE`.